### PR TITLE
Fix 1444924

### PR DESCRIPTION
**Pull request by juliusmilan**
resolves problem with DEBUG messages considered as BUGs in kernel log.
added koops suspicious string blacklist, with one entry yet - "DEBUG"
added test

### CODE REVIEW

File: src/lib/kernel.c [hunk 1]
```diff
@@ -150,6 +150,37 @@ static const char *const s_koops_suspicious_strings[] = {
...
+            in_suspicious_list = true;
+        }
+    }
+
+    bool in_suspicious_blacklist = false;
+    for (const char *const *str = s_koops_suspicious_strings_blacklist; *str; ++str)
+    {
+        if (strstr(line, *str))
+        {
+            in_suspicious_blacklist = true;
```
Comment (jfilak) [inline src/lib/kernel.c]: We can stop break the loop with the first matching blacklist item.

File: src/plugins/abrt-dump-journal-xorg.c [hunk 2]
```diff
@@ -114,6 +114,7 @@ static void watch_journald(abrt_journal_t *journal, const char *dump_location, i
         .decorated_cb = abrt_journal_watch_extract_xorg_crashes,
         .decorated_cb_data = &watch_conf,
         .strings = xorg_strings,
+        .blacklisted_strings = NULL,
```
Comment (jfilak) [inline src/plugins/abrt-dump-journal-xorg.c]: Why do you need to touch the Xorg parsing in commit fixing a bug in the Kernel oops parsing?

File: src/plugins/abrt-journal.h [hunk 3]
```diff
@@ -129,6 +129,7 @@ struct abrt_journal_watch_notify_strings
     abrt_journal_watch_callback decorated_cb;
     void *decorated_cb_data;
     GList *strings;
+    GList *blacklisted_strings;
```
Comment (jfilak) [inline src/plugins/abrt-journal.h]: Can you please create a separate commit for the changes in abrt-journal API?

File: src/plugins/abrt-journal.c [hunk 4]
```diff
@@ -469,15 +469,16 @@ void abrt_journal_watch_notify_strings(abrt_journal_watch_t *watch, void *data)
...
     GList *cur = conf->strings;
-    while (cur)
-    {
+    for (; cur; cur = g_list_next(cur))
         if (strstr(message, cur->data) != NULL)
             break;
 
-        cur = g_list_next(cur);
-    }
+    GList *blacklist_cur = conf->blacklisted_strings;
```
Comment (jfilak) [inline src/plugins/abrt-journal.c]: With no white-list match, there is no need to iterate over the black-list items.

File: src/plugins/abrt-journal.h (hunk 3 above)
Comment (juliusmilan) [inline src/plugins/abrt-journal.h]: done

File: src/lib/kernel.c [hunk 5]
```diff
@@ -150,6 +150,36 @@ static const char *const s_koops_suspicious_strings[] = {
...
+static bool suspicious_line(const char *line)
+{
+    bool in_suspicious_list = false;
+    for (const char *const *str = s_koops_suspicious_strings; *str; ++str)
+        if (strstr(line, *str))
+        {
+            in_suspicious_list = true;
+            break;
+        }
+
```
Comment (jfilak) [inline src/lib/kernel.c]: Just a side note. I believe that fast returns would make the code more readable:
```c
const char *const *str = s_koops_suspicious_strings;
for ( ; *str; ++str)
    if (strstr(line, *str))
        break;

if (!*str)
    return false;

str = s_koops_suspicious_strings_blacklist
for ( ; *str; ++str)
    if (strstr(line, *str))
       break;

return !!*str;
```

File: src/plugins/abrt-dump-journal-xorg.c (hunk 2 above)
Comment (jfilak) [inline src/plugins/abrt-dump-journal-xorg.c]: No need to change anything. This commit makes sense now - I was just confused.

**Comment by mhabrnal** [comment]
Thank you!

**Pull request merged**